/*
 *  Copyright (c) 2020 NetEase Inc.
 *
 *  Licensed under the Apache License, Version 2.0 (the "License");
 *  you may not use this file except in compliance with the License.
 *  You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 *  Unless required by applicable law or agreed to in writing, software
 *  distributed under the License is distributed on an "AS IS" BASIS,
 *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *  See the License for the specific language governing permissions and
 *  limitations under the License.
 */

syntax="proto2";
package curve.chunkserver;
option cc_generic_services = true;

message ChunkServerMetadata {
    required uint32 version = 1;
    required uint32 id = 2;
    required string token = 3;
    required uint32 checksum = 4;
};

message ChunkServerStatusRequest {
}

message ChunkServerStatusResponse {
    required bool copysetLoadFin = 1;
}

service ChunkServerService {
    rpc ChunkServerStatus (ChunkServerStatusRequest) returns (ChunkServerStatusResponse);
};
